code/modules/antagonists/traitor/components/demoraliser.dm

```diff
@@ -0,0 +1,176 @@
+		to_chat(viewer, span_notice("[moods.authority_notification]"))
+		SEND_SIGNAL(viewer, COMSIG_ADD_MOOD_EVENT, moods.mood_category, moods.authority_mood)
+	else
+		to_chat(viewer, span_notice("[moods.crew_notification]"))
+		SEND_SIGNAL(viewer, COMSIG_ADD_MOOD_EVENT, moods.mood_category, moods.crew_mood)
+
+	// You don't get points for looking at your own stuff.
+	if (viewer != owner)
+		SEND_SIGNAL(host, moods.mood_signal, owner)
+
+/**
+ * Returns true if the viewer already has been given feelings, false if they haven't.
+ *
+ * Arguments
+ * * viewer - Whoever just saw the parent.
+ */
+/datum/proximity_monitor/advanced/demoraliser/proc/has_poster_mood_already(mob/living/carbon/human/viewer)
+	var/datum/component/mood/mood = viewer.GetComponent(/datum/component/mood)
+	for(var/i in mood.mood_events)
+		var/datum/mood_event/moodlet = mood.mood_events[i]
+		if (moodlet.category == moods.mood_category)
+			return TRUE
+
+	return FALSE
+
+/**** Moods ****/
+
+/// Mood application categories for this objective
+/// Used to reduce duplicate code for applying moods to players based on their state
+/datum/demoralise_moods/
+	/// Mood category to apply to moods
+	var/mood_category
+	/// Event signal to send when mood is applied
+	var/mood_signal = COMSIG_DEMORALISING_EVENT
+
+	/// Text to display to an antagonist upon receiving this mood
+	var/antag_notification
+	/// Mood datum to apply to an antagonist
+	var/datum/mood_event/antag_mood
+	/// Text to display to a crew member upon receiving this mood
+	var/crew_notification
+	/// Mood datum to apply to a crew member
+	var/datum/mood_event/crew_mood
+	/// Text to display to a head of staff upon receiving this mood
+	var/authority_notification
+	/// Mood datum to apply to a head of staff or security
+	var/datum/mood_event/authority_mood
+
+/**** Poster Moods ****/
+
+#define EVIL_POSTER_MOODLET_CATEGORY "evil poster"
+
+/datum/demoralise_moods/poster
+	mood_category = EVIL_POSTER_MOODLET_CATEGORY
+	antag_notification = "Nice poster."
+	antag_mood = /datum/mood_event/traitor_poster_antag
+	crew_notification = "Wait, is what that poster says true?"
+	crew_mood = /datum/mood_event/traitor_poster_crew
+	authority_notification = "Hey! Who put up that poster?"
+	authority_mood = /datum/mood_event/traitor_poster_auth
+
+/datum/mood_event/traitor_poster_antag
+	description = "I am doing the right thing."
+	mood_change = 2
+	timeout = 2 MINUTES
+	hidden = TRUE
+
+/datum/mood_event/traitor_poster_crew
+	description = "That poster made me feel bad about my job..."
+	mood_change = -2
+	timeout = 2 MINUTES
+	hidden = TRUE
+
+/datum/mood_event/traitor_poster_auth
+	description = "That poster better not be giving the crew any funny ideas..."
+	mood_change = -3
+	timeout = 2 MINUTES
+	hidden = TRUE
+
+#undef EVIL_POSTER_MOODLET_CATEGORY
+
+/**** graffiti Moods ****/
+
+#define EVIL_GRAFFITI_MOODLET_CATEGORY	"evil graffiti"
```

Comment:
ditto, if this is for clarity about what's available, why not stick them in a defines file somewhere